.github/workflows/gradle-nightly-tests.yml

```diff
@@ -0,0 +1,103 @@
+name: Nightly Tests
+
+on:
+  schedule:
+    - cron: "0 22 * * *"
+  workflow_dispatch:
+
+env:
+  CARGO_NET_GIT_FETCH_WITH_CLI: true
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest-128
+    steps:
+      - uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: |
+            ${{ secrets.CONSTRAINTS_SSH_KEY }}
+
+      - name: Checkout repository
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: 21
+          distribution: temurin
+
+      - name: Install Rust
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      # The asset URL for the latest release can be found with:
+      # curl -L -H "Accept: application/vnd.github+json"  \
+      #     -H "Authorization: Bearer YOUR_GH_API_TOKEN" \
+      #     -H "X-GitHub-Api-Version: 2022-11-28" \
+      #     https://api.github.com/repos/ConsenSys/corset/releases/latest
+      # | jq '.assets[] | select(.name|endswith("x86_64-unknown-linux-musl.tar.gz")) | .url'
+      - name: Install Corset
+        run: RUSTFLAGS=-Awarnings cargo install --git ssh://git@github.com/ConsenSys/corset --tag v9.7.13 --locked --force
+#        run: |
+#          curl -L \
+#          -H "Accept: application/octet-stream" \
+#          -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+#          -H "X-GitHub-Api-Version: 2022-11-28" \
+#          -o corset.tar.gz \
+#          https://api.github.com/repos/Consensys/corset/releases/assets/151396061
+#          tar xzf corset.tar.gz
+#          mv corset $HOME
+#          echo $HOME >> $GITHUB_PATH
+
+      - name: Run Nightly tests
+        run: ./gradlew nightlyTests
+        env:
+          JAVA_OPTS: -Dorg.gradle.daemon=false
+          CORSET_FLAGS: fields,expand,expand,expand
+
+      - name: Upload test report
+        if: always()
+        uses: actions/upload-artifact@v3
+        with:
+          name: unit-test-report
+          path: arithmetization/build/reports/tests/
+
+  go-corset-tests:
+    runs-on: ubuntu-latest-128
```

Comment:
do we need this runner or can it use the standard one?

Reply:
Eventually I would like to move all reference tests in this workflow by tagging them as nightly so we need a runner with high memory. 

However for the workflows that run on every commit can use a standard one.